tests/integration_tests/tests/agentless_tests/test_snapshot.py

```diff
@@ -238,11 +244,58 @@ def test_v_5_0_5_restore_snapshot_with_updated_deployment(self):
         self.assertListEqual(plugins_to_uninstall, [])
         self.assertFalse(runtime_only_evaluation)
 
-    def test_v_4_6_0_restore_snapshot_and_restart_services(self):
-        snapshot_path = self._get_snapshot('snap_4_6_0_hello_world.zip')
+    def test_v_5_0_5_restore_snapshot_and_inter_deployment_dependencies(self):
+        snapshot_path = self._get_snapshot(
+            'snap_5.0.5_with_component_openstack.zip')
         self._upload_and_restore_snapshot(snapshot_path)
-        docl.execute('cfy_manager restart --force')
-        self.assertTrue(self._all_services_restarted_properly())
+
+        inter_deployment_dependencies = postgresql.run_query(
+            "SELECT _source_deployment, _target_deployment,  "
+            "dependency_creator FROM inter_deployment_dependencies;"
+        )['all']
+
+        assert (self._openstack_inter_deployment_dependencies()).issubset(
+            set(inter_deployment_dependencies))
+
+        if not self._assert_component_listed(inter_deployment_dependencies):
+            raise AssertionError('component.infrastructure does not exist '
+                                 'as a dependency creator.')
+
+    @staticmethod
+    def _assert_component_listed(inter_deployment_dependencies):
+        for dependency in inter_deployment_dependencies:
+            if dependency[2].startswith('component.infrastructure'):
+                return True
+
+        return False
```

Comment:
could be written as `return any(d[2].startswith(..) for d in inter_dep_deps)`

Reply:
Oh nice, done